Expose PR head ref name to prow jobs

dobesv opened the pull request:
Some tools really want to know the name of the branch being built (chromatic.com for example).

This could also be handy in some places where currently only the PR # is displayed; it would be nice to show the branch name as well as that is generally more user friendly.

This PR a initially a proposal - some more cleanup and testing is likely needed.  I haven't figured out how to build and run these tools myself as a test yet.

cjwagner commented on prow/pod-utils/downwardapi/jobspec.go:
```diff
@@ -98,13 +98,14 @@ const (
+	pullRefsEnv        = "PULL_REFS"
+	pullNumberEnv      = "PULL_NUMBER"
+	pullPullShaEnv     = "PULL_PULL_SHA"
+	pullHeadRefNameEnv = "PULL_HEAD_REF_NAME"
```
How about `PULL_HEAD_REF` for naming consistency?

dobesv replied:
Sounds fine to me, I updated it.

cjwagner reviewed (commented):
Generally looks good, but I think we should be a bit more consistent about naming.

Please also update the env documentation once this is in: https://docs.prow.k8s.io/docs/jobs/#job-environment-variables

dobesv commented:
> Generally looks good, but I think we should be a bit more consistent about naming.

No problem, I updated it as recommended.

> Please also update the env documentation once this is in

Yeah, I can do that once this is in.

I also updated the tests now, so they should pass if you run them.

Let me know if there's anything else I can do!
